### PR TITLE
Made user session store base directory lazy, logging it for easier de…

### PR DESCRIPTION
**Pull request by stefanceriu**
…bugging

### CODE REVIEW

File: ElementX/Sources/Services/UserSessionStore/UserSessionStore.swift [hunk 1]
```diff
@@ -136,7 +136,7 @@ class UserSessionStore: UserSessionStoreProtocol {
         }
     }
     
-    func baseDirectory() -> URL {
+    lazy var baseDirectory: URL = {
```
Comment (pixlwave) [inline ElementX/Sources/Services/UserSessionStore/UserSessionStore.swift:139]: Probably worth making this a `private(set)`

**Review by pixlwave — APPROVED** [review]

File: ElementX/Sources/Services/UserSessionStore/UserSessionStore.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/UserSessionStore/UserSessionStore.swift:139]: Upsie 🙃 
I wonder if we should drop the `baseDirectoryPath` and just keep this one?

File: ElementX/Sources/Services/Authentication/AuthenticationServiceProxy.swift [hunk 2]
```diff
@@ -34,7 +34,7 @@ class AuthenticationServiceProxy: AuthenticationServiceProxyProtocol {
     
     init(userSessionStore: UserSessionStoreProtocol) {
         self.userSessionStore = userSessionStore
-        authenticationService = AuthenticationService(basePath: userSessionStore.baseDirectoryPath)
+        authenticationService = AuthenticationService(basePath: userSessionStore.baseDirectory.path)
```
Comment (pixlwave) [inline ElementX/Sources/Services/Authentication/AuthenticationServiceProxy.swift:37]: Lol not sure why I didn't just do this 🙈

**Review by pixlwave — APPROVED** [review]
LGTM

**Pull request merged**